workflow/controller/retry_tweak.go

```diff
@@ -0,0 +1,23 @@
+package controller
+
+import (
+	wfv1 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+	wfutil "github.com/argoproj/argo/workflow/util"
+)
+
+type RetryTweak interface {
+	RetryTweak(wfv1.RetryStrategy, wfv1.Nodes, *wfv1.Template)
+}
+
+type RetryOnDifferentHost struct {
```

Comment:
with my idea, this would need to be a 2nd order function, e.g.

```golang
func RetryOnDifferentHost(retryHostName string) RetryTweak {
  return func(retryStrategy wfv1.RetryStrategy, allNodes wfv1.Nodes, tmpl *wfv1.Template) {
	hostNames := wfutil.GetFailHosts(allNodes, r.retryNodeName)
	hostLabel := retryStrategy.ScheduleOnDifferentHostNodesLabel
	if hostLabel != nil && len(hostNames) > 0 {
		tmpl.Affinity = wfutil.AddHostnamesToAffinity(*hostLabel, hostNames, tmpl.Affinity)
	}
}
```